packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -357,15 +358,21 @@ export class Nodegroup extends Resource implements INodegroup {
       Annotations.of(this).addWarning('"instanceType" is deprecated and will be removed in the next major version. please use "instanceTypes" instead');
     }
     const instanceTypes = props.instanceTypes ?? (props.instanceType ? [props.instanceType] : undefined);
-    let expectedAmiType = undefined;
+    let possibleAmiTypes: NodegroupAmiType[] = [];
 
     if (instanceTypes && instanceTypes.length > 0) {
-      // if the user explicitly configured instance types, we can calculate the expected ami type.
-      expectedAmiType = getAmiType(instanceTypes);
-
-      // if the user explicitly configured an ami type, make sure its the same as the expected one.
-      if (props.amiType && props.amiType !== expectedAmiType) {
-        throw new Error(`The specified AMI does not match the instance types architecture, either specify ${expectedAmiType} or dont specify any`);
+      /**
+       * if the user explicitly configured instance types, we can't caculate the expected ami type as we support
+       * Amazon Linux 2 and Bottlerocket now. However we can check:
```

Comment:
Could we do the opposite? Take the AMI type as a starting point (if it's provided) and figure out which instance types are valid for it?

Reply:
sounds like good, wdyt @pahud.

Reply:
@otaviomacedo Yes this is exactly what I am working on.

I introduced `possibleAmiTypes` in this PR and calculate from the provided instance types.

Let's say if you provide all x86 instance types the `possibleAmiTypes` would only be ` AL2_X86_64` and `BOTTLEROCKET_X86_64`. In this case if `amiType` is provided we can check if it's included in the `possibleAmiTypes`.